npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater.rb

```diff
@@ -63,11 +64,11 @@ def updated_dependency_files
 
       def vendor_updated_files(updated_files)
         base_dir = updated_files.first.directory
-        pnp_updater.updated_vendor_cache_files(base_directory: base_dir).each do |file|
-          updated_files << file if file.name == ".pnp.cjs" || file.name == ".pnp.data.json"
+        pnp_updater.updated_files(base_directory: base_dir, only_paths: [".pnp.cjs", ".pnp.data.json"]).each do |file|
+          updated_files << file
```

Comment:
This was an opportunistic change to allow us to optimise very slightly to avoid reading any files we don't plan on using I've made in passing.

I'm ok with backing this out as it _is_ scope creep, but it was a fairly quick chance to make in passing.